src/v/model/fundamental.h

```diff
@@ -198,7 +199,7 @@ using control_record_version
 
 static constexpr control_record_version current_control_record_version{0};
 
-enum class shadow_indexing_mode : int8_t {
+enum class shadow_indexing_mode : uint8_t {
```

Comment:
nit: i think it would be much easier to use bit flags approach like we did with `cleanup_policy_bitflags`.

Reply:
I switched to `std::underlying_type_t`. 
I don't like that the shadow_indexing_mode is a single field. Going to refactor this code when we will integrate serde. This code would be much simpler with two fields.

Reply:
got it thank you for explaining this.